### PR TITLE
*: simplify record/index key codec.

### PR DESCRIPTION
**Pull request by siddontang**
@qiuyesuifeng  @disksing

### CODE REVIEW

**Comment by qiuyesuifeng** [comment]
LGTM

File: kv/index_iter.go [hunk 1]
```diff
@@ -118,11 +117,7 @@ func NewKVIndex(indexPrefix string, indexName string, indexID int64, unique bool
 		unique:    unique,
 	}
 
-	var err error
-	index.prefix, err = GenIndexPrefix(indexPrefix, indexID)
-	if err != nil {
-		return nil, errors.Trace(err)
-	}
+	index.prefix = GenIndexPrefix(indexPrefix, indexID)
```
Comment (disksing) [inline kv/index_iter.go]: We can generate prefix in struct literal directly.

File: table/tables/tables.go [hunk 2]
```diff
@@ -753,47 +735,85 @@ func GetColDefaultValue(ctx context.Context, col *model.ColumnInfo) (interface{}
...
 
-// EncodeRecordKey encodes the string value to a byte slice.
-func EncodeRecordKey(tablePrefix string, h int64, columnID int64) []byte {
-	var (
-		buf []byte
-		err error
-	)
+var (
+	recordPrefixSep = []byte("_r")
+	indexPrefixSep  = []byte("_i")
```
Comment (disksing) [inline table/tables/tables.go]: Is the '_' character here necessary or just for readability?

Comment (siddontang) [inline table/tables/tables.go]: old source, let keep it here now. 
It is not hard to refactor later.

Comment (disksing) [inline table/tables/tables.go]: Got it. LGTM.

**Pull request merged**